### DISCUSSION TITLE
type [2147483647]byte larger than address space

### DISCUSSION THREAD

**Discussion by aymentrosh** [ethereum-optimism/developers#238] [Deploying the OP Stack]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

when i run `make devnet-up` after sometime i get this error 
```
failed to solve: process "/bin/sh -c cd op-program && make op-program-host      GOOS=$TARGETOS GOARCH=$TARGETARCH GITCOMMIT=$GIT_COMMIT GITDATE=$GIT_DATE VERSION=\"$OP_PROGRAM_VERSION\"" did not complete successfully: exit code: 2
```
and when I run this command manually `make op-program `
I get this issue 
```
type [2147483647]byte larger than address space
make: *** [op-program-client-mips] Error 1
```
any help?

**Comment by aymentrosh** [comment]
I fixed that and now when I run `make devnet-up` it runs but it keeps returning this 
```
[INFO|02-09-2024 11:06:30] Waiting for RPC server at 127.0.0.1:9545
[INFO|02-09-2024 11:06:31] Waiting for RPC server at 127.0.0.1:9545
[INFO|02-09-2024 11:06:32] Waiting for RPC server at 127.0.0.1:9545
[INFO|02-09-2024 11:06:33] Waiting for RPC server at 127.0.0.1:9545
[INFO|02-09-2024 11:06:34] Waiting for RPC server at 127.0.0.1:9545
[INFO|02-09-2024 11:06:35] Waiting for RPC server at 127.0.0.1:9545
```

**Reply by hash1go** [reply]
Are you running the RPC server on a local server?
If so, please ensure that TCP port 9545 is not blocked by a firewall or similar.

**Reply by aymentrosh** [reply]
fixed that that was an issue with op-geth didn't start and it keeps waiting for him 
for now i have another issue is op-node op-batcher op-proposer returning the same issue 
`2024-02-09 14:37:45 exec /usr/local/bin/op-node: exec format error`
inside docker

**Reply by hash1go** [reply]
Please make sure that the Docker image you used matches the CPU architecture (x86 or ARM or ...).

**Reply by aymentrosh** [reply]
I'm not sure why I'm using this [dockerfile](https://github.com/ethereum-optimism/optimism/blob/op-batcher/v1.4.3/op-node/Dockerfile)

**Reply by hash1go** [reply]
Do you know the CPU architecture of the machine you are using?

**Reply by aymentrosh** [reply]
I'm using a MacBook Air chip M2

**Reply by hash1go** [reply]
Since you are using a MacBook Air with an ARM architecture, it's possible that the Docker image you are trying to use is not compatible with your system.
I recommend building the application directly from the source on your machine.
https://docs.optimism.io/builders/node-operators/tutorials/mainnet

Also, you can consider using a machine with a CPU of the more commonly prevalent x86 architecture.

**Comment by sbvegan** [comment]
Hey @aymentrosh, the devnet was (might still be) getting changes last week. Can you try again and let us know if you're getting the same error? I also recommend making sure all your dependencies are up to date.
